test/Microsoft.IdentityModel.Tokens.Tests/EventBasedLRUCacheTests.cs

```diff
@@ -105,10 +105,8 @@ public void CompactCache()
             AddItemsToCache(cache, 3 * capacity, expiredInSeconds);
 
             // allows the event queue task to be started to compact the cache
-            Thread.Sleep(100);
-
-            // wait until the event queue is empty
             cache.WaitForProcessing();
+            Thread.Sleep(waitInMiliSeconds);
```

Comment:
These tests in general have an awful lot of Thread.Sleep calls

Is there something that prevents us from hooking into the LRU cache and getting an event based hook or at least an override for expiry time? Sleeps of 2 seconds here, 5 seconds in the above test really start to add up, especially if we're running tests serially.

Reply:
That is a valid point, while it is not straightforward to predict a thread action, it is a bad practice to hardcode a wait time because it increases the build time. This should be fixed now.